tests/network-tests/src/fixtures/referendum/voteFixture.ts

```diff
@@ -0,0 +1,68 @@
+import { SubmittableExtrinsic } from '@polkadot/api/types'
+import { ISubmittableResult } from '@polkadot/types/types'
+import BN from 'bn.js'
+import { assert } from 'chai'
+import { Api } from '../../Api'
+import { StandardizedFixture } from '../../Fixture'
+import { QueryNodeApi } from '../../QueryNodeApi'
+import { EventDetails, AnyQueryNodeEvent } from '../../types'
+
+export type VotingParams = {
+  commitment: string
+  stake: BN
+}
+
+export class VoteFixture extends StandardizedFixture {
+  private votes: Map<string, VotingParams>
+  private failureExpected: boolean
+
+  constructor(api: Api, query: QueryNodeApi, votes: Map<string, VotingParams>, failureExpected = false) {
+    super(api, query)
+    this.votes = votes
+    this.failureExpected = failureExpected
+    this.decrementalTip = true
+  }
+
+  protected getSignerAccountOrAccounts(): Promise<string | string[]> {
+    return Promise.resolve(Array.from(this.votes.keys()))
+  }
+
+  protected async getExtrinsics(): Promise<SubmittableExtrinsic<'promise', ISubmittableResult>[]> {
+    return Array.from(this.votes).map(([, params]) => this.api.tx.referendum.vote(params.commitment, params.stake))
+  }
+
+  protected getEventFromResult(result: ISubmittableResult): Promise<EventDetails<unknown>> {
+    return this.api.getEventDetails(result, 'referendum', 'VoteCast')
+  }
+
+  protected assertQueryNodeEventIsValid(qEvent: AnyQueryNodeEvent, i: number): void {
+    // TODO: implement QN checks
+  }
+
+  // Due to the fact that we need the transactions to be processed in the expected order
+  // (which is not guaranteed by the nonce, because we're using different voter accounts),
+  // we'll be including a small, decremental tip (10 JOY * (votersStakingAccounts.length - 1 - accIndex))
+  public async execute(): Promise<void> {
+    const accountOrAccounts = await this.getSignerAccountOrAccounts()
+    const extrinsics = await this.getExtrinsics()
+    this.extrinsics = extrinsics.flat()
+    await this.api.prepareAccountsForFeeExpenses(accountOrAccounts, this.extrinsics, this.decrementalTip ? 10 : 0)
+    this.results = await this.api.sendExtrinsicsAndGetResults(
+      extrinsics,
+      accountOrAccounts,
+      this.decrementalTip ? 10 : 0
+    )
+    if (!this.failureExpected) {
```

Comment:
I think it could be a good idea to also make expected failure a configurable part of `StandardizedFixture` aswell.

Actually now that I think about it, you could even do the following to make thing much simpler:
1. Add a method like `setExpectedError(errorName)` to `StandardizedFixture`, which would set `this.expectedError` to `errorName` and accordingly change the logic of `execute` if `this.expectedError` is set (something like `if (this.expectedError) { this.results.forEach((r) => this.expectDispatchError(r, this.expectedError) }`
2. Make `votersStakingAccounts` optionally configurable in `ElectCouncilFixture`
3. Add `setExpectedVotingError(errorName)` to `ElectCouncilFixture` which will set `this.expectedVotingError` and call `votingFixture.expectFailure(this.expectedVotingError)` if `this.expectedVotingError` is set.
4. You now don't need a separate fixture for voting opt-out test. You can just:
    - Create voter keys
    - Run `BlacklistVoteFixture`
    - Create `ElectCouncilFixture` via `const electCouncilFixture = new ElectCouncilFixture(voterKeys).setExpectedVotingError('AccountAlreadyOptedOutOfVoting')`
    - Run the fixture

I think this will be a very nice solution, as `VotersOptingOut` fixture by itself is not very reusable, it just represents a single test case and duplicates a lot of `ElectCouncilFixture` code.

Additionally I think introducing some standarized way of testing for expected failures can possibly pay off a lot in the future.